src/plone/api/tests/test_content.py

```diff
@@ -610,6 +616,81 @@ def test_delete_multiple(self):
         assert 'copy_of_about' not in container
         assert 'about' not in container['events']
 
+    def test_delete_ignore_linkintegrity(self):
+        """Test deleting a content item with a link pointed at it."""
+        self._set_text(self.team, '<a href="contact">contact</a>')
+        # Delete the contact page
+        api.content.delete(self.contact, check_linkintegrity=False)
+        assert 'contact' not in self.portal['about'].keys()
+
+    def test_delete_check_linkintegrity(self):
+        """Test deleting a content item with a link pointed at it."""
+        if not IBaseContent.providedBy(self.team) or not NEW_LINKINTEGRITY:
+            # This test only makes sense with Archetypes or new Linkintegrity
+            return
+        self._set_text(self.team, '<a href="contact">contact</a>')
+        # Delete the contact page
+        with self.assertRaises(LinkIntegrityNotificationException):
+            api.content.delete(self.contact)
+        assert 'contact' in self.portal['about'].keys()
+        if NEW_LINKINTEGRITY:
+            # In the old implementation of linkintegrity the items are
+            # still gone during this request.
+            self.assertIn('contact', self.portal['about'].keys())
+
+    def test_delete_multiple_check_linkintegrity(self):
+        """Test deleting multiple item with linkintegrity-breaches."""
+        if not IBaseContent.providedBy(self.team) or not NEW_LINKINTEGRITY:
+            # This test only makes sense with Archetypes or new Linkintegrity
+            return
+        self._set_text(self.team, '<a href="../about/contact">contact</a>')
+        self._set_text(self.training, '<a href="../blog">contact</a>')
+        # Delete the contact page
+        with self.assertRaises(LinkIntegrityNotificationException):
+            api.content.delete(objects=[self.blog, self.contact])
+        if NEW_LINKINTEGRITY:
+            # In the old implementation of linkintegrity the items are
+            # still gone during this request.
+            self.assertIn('contact', self.portal['about'].keys())
+            self.assertIn('blog', self.portal.keys())
+
+    def test_delete_multiple_ignore_linkintegrity(self):
+        """Test deleting multiple items ignoring linkintegrity-breaches."""
+        if not IBaseContent.providedBy(self.team) or not NEW_LINKINTEGRITY:
```

Comment:
In the decorator ``@skipIf`` I have no access to ``self``, so I cannot test of ``self.team`` is Archetypes or Dexterity. 
Instead I could after the imports if ``plone.app.contenttypes`` is installed which would mean it is Dexterity.